### PR TITLE
Added broadcast and events for app images and exports.

### PR DESCRIPTION
**Pull request by paulnexuspoint**

### CODE REVIEW

**Review by npharry — DISMISSED** [review]

File: app/Broadcasts/LogSmash/App/LogAppImageStoreRequestBroadcast.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
+<?php
```
Comment (peterfrankjohnson) [inline app/Broadcasts/LogSmash/App/LogAppImageStoreRequestBroadcast.php:1]: Sorry to be a pedant, but App really doesn't describe this at all.

Can it not be called "PhotoApp" or "ImageApp" or something? There will be a day when we write another App so they should be differentiated.

**Review by sroberts-coder — DISMISSED** [review]

File: app/Broadcasts/LogSmash/App/LogAppImageStoreRequestBroadcast.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    /**
+     * Get the channels the event should be broadcast on.
+     *
+     * @return array
+     */
+    public function broadcastOn()
+    {
+        return ['log-smash.api.' . $this->tenant['guid']];
```
Comment (peterfrankjohnson) [inline app/Broadcasts/LogSmash/App/LogAppImageStoreRequestBroadcast.php]: Also, this needs a unique name as well.

Something like log-smash.photo-app or whatever

File: app/Broadcasts/LogSmash/Export/LogExportBroadcast.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    /**
+     * Get the channels the event should be broadcast on.
+     *
+     * @return array
+     */
+    public function broadcastOn()
+    {
+        return ['log-smash.api.' . $this->tenant['guid']];
```
Comment (peterfrankjohnson) [inline app/Broadcasts/LogSmash/Export/LogExportBroadcast.php]:
```suggestion
        return ['log-smash.export.' . $this->tenant['guid']];
```

File: app/Broadcasts/LogSmash/Export/LogExportBroadcast.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        string $method,
+        array $request
+    ) {
+        $this->guid        = $guid;
+        $this->environment = $environment;
+        $this->timestamp   = $timestamp;
+        $this->tenant      = $tenant;
+        $this->type        = $type;
+        $this->method      = $method;
+        $this->request     = $request;
```
Comment (peterfrankjohnson) [inline app/Broadcasts/LogSmash/Export/LogExportBroadcast.php]:
```suggestion
```

These fields aren't relevant to exports, these were specifically for API requests.

There need to be fields for the export, such as export name, stage (of the export process), etc...

File: app/Broadcasts/LogSmash/App/LogAppImageStoreRequestBroadcast.php (hunk 1 above)
Comment (paulnexuspoint) [reply app/Broadcasts/LogSmash/App/LogAppImageStoreRequestBroadcast.php:1]: I had originally thought of that, but the current app is being extended to cover a number of different things. This app is currently referred to as the Apex App, so I believe subsequent apps would be named accordingly like Apex Pin The Tail On The Donkey App. In this case the current app will always be the main app and others will be children apps.

**Review by peterfrankjohnson — APPROVED** [review]

**Review by sroberts-coder — APPROVED** [review]

**Review by npharry — APPROVED** [review]

**Review by matthans0n — APPROVED** [review]

**Pull request merged**